src/state/actions/auth.js

```diff
@@ -234,3 +240,82 @@ export const changeUserPassword = (currentPassword, newPassword) => {
     return dispatch(AUTH_CHANGE_PASSWORD_SUCCESS());
   };
 };
+
+const authWithProvider = provider => {
+  return async (dispatch, getState) => {
+    dispatch(AUTH_PROVIDER_INIT());
+    const { locale } = getState().preferences;
+    let result;
+
+    try {
+      result = await firebase.auth().signInWithPopup(provider);
+    } catch (error) {
+      const errorMessage = firebaseError(error.code, locale);
+      return dispatch(AUTH_PROVIDER_FAIL({ error: errorMessage }));
+    }
+    const { user, additionalUserInfo } = result;
+
+    const { uid } = firebase.auth().currentUser;
+
+    const createdAt = new Date().toString();
+
+    const { location } = additionalUserInfo.profile;
+
+    const userData = {
+      isAdmin: false,
+      email: user.email,
+      name: user.displayName,
+      createdAt,
+      logoUrl: user.photoURL,
+      location: location ? location.name : null
+    };
+
+    let userValue;
```

Comment:
The same with this variable. We can name it like `userFromDb` to give an example.